src/index.js

```diff
@@ -130,11 +130,14 @@ function raw (service, method, params) {
   // handle client methods like indices.create
   const [meth, ext] = method.split('.');
 
-  if (typeof ext !== 'undefined') {
-    return service.Model[meth][ext](params);
+  if (typeof service.Model[meth] === 'undefined') {
+    return Promise
+      .reject(new Error(`There is no query method ${meth}`));
   }
 
-  return service.Model[meth](params);
+  return (typeof service.Model[meth][ext] === 'function')
```

Comment:
The way I read it, when `ext` is defined but is not a function we will silently fall back to running `Model[meth]` (btw, `meth` seems to be quite an unfortunate name for a variable).

I believe we should be returning rejected promise when `ext` is defined but not a function. In this scenario the user tried to run a function which does not exist and we can actually do something he did not intend at all.